modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/utils/DASConfigurationUtils.java

```diff
@@ -0,0 +1,287 @@
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.core.service.RegistryService;
+import org.wso2.carbon.registry.core.session.UserRegistry;
+import org.wso2.carbon.utils.CarbonUtils;
+import org.wso2.securevault.SecretResolver;
+import org.wso2.securevault.SecretResolverFactory;
+
+import javax.xml.bind.DatatypeConverter;
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.transform.TransformerException;
+import java.io.File;
+import java.io.IOException;
+
+import org.wso2.carbon.pc.core.ProcessStore;
+
+/**
+ * Utils file for Analytics configuration related properties
+ */
+public class DASConfigurationUtils {
+    private static final Log log = LogFactory.getLog(DASConfigurationUtils.class);
+
+    /**
+     * Check whether analytics configurations (with DAS) is made for the respective process
+     *
+     * @param processName
+     * @param processVersion
+     * @return isDASAnalyticsConfigured
+     * @throws ProcessCenterException
+     */
+    public static boolean isDASAnalyticsConfigured(String processName, String processVersion)
+            throws ProcessCenterException {
+        String processContent = null;
+        ProcessStore ps = new ProcessStore();
+        try {
+            RegistryService registryService = PCAnalyticsServerHolder.getInstance().getRegistryService();
+            if (registryService != null) {
+                UserRegistry reg = registryService.getGovernanceSystemRegistry();
+                String processAssetPath = ProcessCenterConstants.PROCESS_ASSET_ROOT + processName + "/" +
+                        processVersion;
+                org.wso2.carbon.registry.core.Resource resource = reg.get(processAssetPath);
+                processContent = new String((byte[]) resource.getContent());
+                Document doc = ps.stringToXML(processContent);
+
+                Element rootElement = doc.getDocumentElement();
+                Element propertiesElement = (Element) rootElement.getElementsByTagName("properties").item(0);
+
+                if (propertiesElement.getElementsByTagName(AnalyticsConfigConstants.IS_DAS_CONFIGED_TAG).getLength() > 0
+                        && propertiesElement.getElementsByTagName(AnalyticsConfigConstants.IS_DAS_CONFIGED_TAG).item(0)
+                        .getTextContent().equals("true")) {
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+        } catch (RegistryException e) {
+            String errMsg = "Governance Registry access error, wile checking isDasConfigedForAnalytics";
+            throw new ProcessCenterException(errMsg, e);
+        } catch (Exception e) {
+            String errMsg = "Registry process.rxt String to XML conversion error, wile checking "
+                    + "isDasConfigedForAnalytics ";
+            throw new ProcessCenterException(errMsg, e);
```

Comment:
This can be any exception, not only xml conversion. Have to add a generic error message.